docs/src/main/tut/auth.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: default
+title: Authentication
+---
+## Authentication
+
+A [service] is a `Kleisli[Task, Request, Response]`, the composable version of
+`Request => Task[Response]`. http4s provides an alias called
+`Service[Request, Response]`. A service with authentication also requires some
+kind of `User` object which identifies which user did the request. So the
+service has the signature `(User, Request) => Task[Response]`, or
+`Service[(User, Request), Response]`. So we'll need a `Request => User`
+function, or more likely, a `Request => Task[User]`, because the `User` will
+come from a database. Which is a `Service[Request, User]`, which we can compose
+with the `Service[(User, Request), Response]` to get a [service]. Or in code:
+
+```tut:book
+import scalaz._, Scalaz._, scalaz.concurrent.Task
+import org.http4s._
+
+case class User(id: Long, name: String)
+
+def addInput[I, T](k: Service[I, T]): Service[I, (T, I)] =
+  Kleisli(input => k.run(input).map(user => (user, input)))
+
+val authUser: Service[Request, User] = Kleisli(_ => Task.delay(???))
+val authedService: Service[(User, Request), Response] = Kleisli(_ => Task.delay(???))
```

Comment:
Would be nice to know what this looks like.  If using http4s-dsl, I guess:

```scala
Service { // Not HTTP service anymore!
  case (user, GET -> Root / "protected" / blah) => Ok("blah")
}
```

That shows how the DSL composes along with the Kleisli.